### ISSUE TITLE
CRC 1.33.1 After CRC installation, CRC restart give issues

### ISSUE THREAD

**Issue by rupeshad81** [code-ready/crc#2791]
### General information

  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / Hyper-V / hyperkit
  * Did you run `crc setup` before starting it (Yes/No)?
  * Running CRC on: Laptop / Baremetal-Server / VM

## CRC version
```bash
# Put `crc version` output here
[crc@eaas-crc ~]$ crc version
CodeReady Containers version: 1.33.1+3cfaa10f
OpenShift version: 4.8.12 (embedded in executable)
[crc@eaas-crc ~]$

  
## CRC status
```bash
# Put `crc status --log-level debug` output here
[crc@eaas-crc ~]$ crc status --log-level debug
DEBU CodeReady Containers version: 1.33.1+3cfaa10f
DEBU OpenShift version: 4.8.12 (embedded in executable)
DEBU Running 'crc status'
DEBU Checking file: /home/crc/.crc/machines/crc/.crc-exist
DEBU Checking file: /home/crc/.crc/machines/crc/.crc-exist
DEBU Found binary path at /home/crc/.crc/bin/crc-driver-libvirt
DEBU Launching plugin server for driver libvirt
DEBU Plugin server listening at address 127.0.0.1:45496
DEBU () Calling .GetVersion
DEBU Using API Version 1
DEBU () Calling .SetConfigRaw
DEBU () Calling .GetMachineName
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-05T12:54:28+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:54:28+01:00" level=debug msg="Fetching VM..."
DEBU (crc) Calling .GetBundleName
DEBU (crc) Calling .GetIP
DEBU (crc) DBG | time="2021-10-05T12:54:28+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:54:28+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:54:28+01:00" level=debug msg="IP address: 192.168.130.11"
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1
DEBU Using ssh private keys: [/home/crc/.crc/machines/crc/id_ecdsa /home/crc/.crc/cache/crc_libvirt_4.8.12/id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output: 354860118016 9323663360 /sysroot
DEBU cannot get OpenShift status: Get "https://api.crc.testing:6443/apis/config.openshift.io/v1/clusteroperators": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU Successfully made call to close driver server
DEBU (crc) DBG | time="2021-10-05T12:54:33+01:00" level=debug msg="Closing plugin on server side"
DEBU Making call to close connection to plugin binary
CRC VM:          Running
OpenShift:       Unreachable (v4.8.12)
Disk Usage:      9.324GB of 354.9GB (Inside the CRC VM)
Cache Usage:     16.06GB
Cache Directory: /home/crc/.crc/cache


## CRC config
```bash
# Put `crc config view` output here
[crc@eaas-crc ~]$ crc config view
- consent-telemetry                     : yes
- http-proxy                            : http://10.232.233.70:8080
- https-proxy                           : http://10.232.233.70:8080
- no-proxy                              : .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing
- proxy-ca-file                         : /tmp/amdocs-certificat.crt


## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
[crc@eaas-crc ~]$ cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.8 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.8"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.8:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.8"


### Steps to reproduce

  1. 
  2. 
  3. 
  4. 

### Expected


### Actual


### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

After the crc successful installation , when we do crc stop and start , it get failed.

crc start --cpus 12 --memory 40000 --log-level debug
DEBU CodeReady Containers version: 1.33.1+3cfaa10f
DEBU OpenShift version: 4.8.12 (embedded in executable)
DEBU Running 'crc start'
DEBU Total memory of system is 42020716544 bytes
DEBU No new version available. The latest version is 1.33.1
DEBU Found binary path at /home/crc/.crc/bin/crc-driver-libvirt
DEBU Launching plugin server for driver libvirt
DEBU Plugin server listening at address 127.0.0.1:39196
DEBU () Calling .GetVersion
DEBU Using API Version 1
DEBU () Calling .SetConfigRaw
DEBU () Calling .GetMachineName
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Fetching VM..."
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Closing plugin on server side"
DEBU Checking if systemd-resolved.service is running
DEBU Running 'systemctl status systemd-resolved.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit systemd-resolved.service could not be found.
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
DEBU Running '/home/crc/.crc/bin/crc-admin-helper-linux --version'
DEBU Found crc-admin-helper-linux version 0.0.6
DEBU crc-admin-helper executable already cached
INFO Checking for obsolete admin-helper executable
DEBU Checking if an older admin-helper executable is installed
DEBU No older admin-helper executable found
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
DEBU Total memory of system is 42020716544 bytes
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
DEBU Checking if the vmx/svm flags are present in /proc/cpuinfo
DEBU CPU virtualization flags are good
INFO Checking if KVM is enabled
DEBU Checking if /dev/kvm exists
DEBU /dev/kvm was found
INFO Checking if libvirt is installed
DEBU Checking if 'virsh' is available
DEBU 'virsh' was found in /usr/bin/virsh
DEBU Checking 'virsh capabilities' for libvirtd/qemu availability
DEBU Running 'virsh --readonly --connect qemu:///system capabilities'
DEBU Found x86_64 hypervisor with 'hvm' capabilities
INFO Checking if user is part of libvirt group
DEBU Checking if current user is part of the libvirt group
DEBU Current user is already in the libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
DEBU Checking if libvirtd service is running
DEBU Running 'systemctl status virtqemud.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.socket could not be found.
DEBU virtqemud.socket is neither running nor listening
DEBU Running 'systemctl status libvirtd.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit libvirtd.socket could not be found.
DEBU libvirtd.socket is neither running nor listening
DEBU Running 'systemctl status virtqemud.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.service could not be found.
DEBU virtqemud.service is neither running nor listening
DEBU Running 'systemctl status libvirtd.service'
DEBU libvirtd.service is running
INFO Checking if a supported libvirt version is installed
DEBU Checking if libvirt version is >=3.4.0
DEBU Running 'virsh -v'
INFO Checking if crc-driver-libvirt is installed
DEBU Checking if crc-driver-libvirt is installed
DEBU Running '/home/crc/.crc/bin/crc-driver-libvirt version'
DEBU Found crc-driver-libvirt version 0.13.1
DEBU crc-driver-libvirt is already installed
INFO Checking if systemd-networkd is running
DEBU Checking if systemd-networkd.service is running
DEBU Running 'systemctl status systemd-networkd.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit systemd-networkd.service could not be found.
DEBU systemd-networkd.service is not running
INFO Checking if NetworkManager is installed
DEBU Checking if 'nmcli' is available
DEBU 'nmcli' was found in /usr/bin/nmcli
INFO Checking if NetworkManager service is running
DEBU Checking if NetworkManager.service is running
DEBU Running 'systemctl status NetworkManager.service'
DEBU NetworkManager.service is already running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
DEBU Checking NetworkManager configuration
DEBU NetworkManager configuration is good
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
DEBU Checking dnsmasq configuration
DEBU dnsmasq configuration is good
INFO Checking if libvirt 'crc' network is available
DEBU Checking if libvirt 'crc' network exists
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU Checking if libvirt 'crc' definition is up to date
DEBU Running 'virsh --connect qemu:///system net-dumpxml --inactive crc'
DEBU libvirt 'crc' network has the expected value
INFO Checking if libvirt 'crc' network is active
DEBU Checking if libvirt 'crc' network is active
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU libvirt 'crc' network is already active
DEBU Checking file: /home/crc/.crc/machines/crc/.crc-exist
DEBU Found binary path at /home/crc/.crc/bin/crc-driver-libvirt
DEBU Launching plugin server for driver libvirt
DEBU Plugin server listening at address 127.0.0.1:34995
DEBU () Calling .GetVersion
DEBU Using API Version 1
DEBU () Calling .SetConfigRaw
DEBU () Calling .GetMachineName
DEBU (crc) Calling .GetBundleName
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Fetching VM..."
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
DEBU Updating CRC VM configuration
DEBU (crc) Calling .GetConfigRaw
DEBU (crc) Calling .Start
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Starting VM crc"
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Validating network"
DEBU (crc) DBG | time="2021-10-05T12:49:25+01:00" level=debug msg="Validating storage pool"
DEBU (crc) DBG | time="2021-10-05T12:49:32+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:32+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:32+01:00" level=debug msg="Waiting for machine to come up 0/60"
DEBU (crc) DBG | time="2021-10-05T12:49:35+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:35+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:35+01:00" level=debug msg="Waiting for machine to come up 1/60"
DEBU (crc) DBG | time="2021-10-05T12:49:38+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:38+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:38+01:00" level=debug msg="Waiting for machine to come up 2/60"
DEBU (crc) DBG | time="2021-10-05T12:49:41+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:41+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:41+01:00" level=debug msg="Waiting for machine to come up 3/60"
DEBU (crc) DBG | time="2021-10-05T12:49:44+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:44+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:44+01:00" level=debug msg="Waiting for machine to come up 4/60"
DEBU (crc) DBG | time="2021-10-05T12:49:47+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:47+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:47+01:00" level=debug msg="Waiting for machine to come up 5/60"
DEBU (crc) DBG | time="2021-10-05T12:49:50+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:50+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:50+01:00" level=debug msg="Waiting for machine to come up 6/60"
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="IP address: 192.168.130.11"
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=info msg="Found IP for machine: 192.168.130.11"
DEBU (crc) Calling .GetConfigRaw
DEBU Waiting for machine to be running, this may take a few minutes...
DEBU retry loop: attempt 0
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="Getting current state..."
DEBU Machine is up and running!
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="Getting current state..."
DEBU (crc) Calling .GetIP
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-05T12:49:53+01:00" level=debug msg="IP address: 192.168.130.11"
INFO CodeReady Containers instance is running with IP 192.168.130.11
DEBU Waiting until ssh is available
DEBU retry loop: attempt 0
DEBU Running SSH command: exit 0
DEBU Using ssh private keys: [/home/crc/.crc/cache/crc_libvirt_4.8.12/id_ecdsa_crc /home/crc/.crc/machines/crc/id_ecdsa]
DEBU SSH command results: err: <nil>, output:
INFO CodeReady Containers VM is running
DEBU Running SSH command: cat /home/core/.ssh/authorized_keys
DEBU SSH command results: err: <nil>, output: ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBACBWAJWCXF38kJpADA+D5KvZP600TuT5F00Sv9Y8wc7Bu9aIfyRnNrvuyoqTkQXFnk7fxQd28o8tF+sJe0IaB3SdQD2DJ+CuzKfEt2xGhEHIw7c9UiWypQmHnmkVUFPv8gGfn0ndSJ+DxCHzK6InHPF4ZUHM+Zfq8NeJYUR/sAdFIKYCg==
DEBU Running SSH command: realpath /dev/disk/by-label/root
DEBU SSH command results: err: <nil>, output: /dev/vda4
DEBU Using root access: Growing /dev/vda4 partition
DEBU Running SSH command: sudo /usr/bin/growpart /dev/vda 4
DEBU SSH command results: err: Process exited with status 1, output: NOCHANGE: partition 4 is size 693106655. it cannot be grown
DEBU No free space after /dev/vda4, nothing to do
DEBU Using root access: make root Podman socket accessible
DEBU Running SSH command: sudo chmod 777 /run/podman/ /run/podman/podman.sock
DEBU SSH command results: err: <nil>, output:
DEBU Creating /var/srv/dnsmasq.conf with permissions 0644 in the CRC VM
DEBU Running SSH command: <hidden>
DEBU SSH command succeeded
DEBU Running SSH command: systemctl status crc-dnsmasq.service
DEBU SSH command results: err: Process exited with status 3, output: â crc-dnsmasq.service - Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service
   Loaded: loaded (/etc/systemd/system/crc-dnsmasq.service; enabled; vendor preset: disabled)
   Active: activating (start) since Tue 2021-10-05 11:49:53 UTC; 1s ago
     Docs: man:podman-generate-systemd(1)
Cntrl PID: 1764 (podman)
    Tasks: 11 (limit: 250732)
   Memory: 50.6M
      CPU: 181ms
   CGroup: /system.slice/crc-dnsmasq.service
           ââ1764 /usr/bin/podman start 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343

Oct 05 11:49:53 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
DEBU Using root access: Executing systemctl enable crc-dnsmasq.service
DEBU Running SSH command: sudo systemctl enable crc-dnsmasq.service
DEBU SSH command results: err: <nil>, output:
DEBU Using root access: Executing systemctl daemon-reload command
DEBU Running SSH command: sudo systemctl daemon-reload
DEBU SSH command results: err: <nil>, output:
DEBU Using root access: Executing systemctl start crc-dnsmasq.service
DEBU Running SSH command: sudo systemctl start crc-dnsmasq.service
DEBU SSH command results: err: Process exited with status 1, output:
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
Error running post start: Executing systemctl action failed:  ssh command error:
command : sudo systemctl start crc-dnsmasq.service
err     : Process exited with status 1
: Job for crc-dnsmasq.service failed because the control process exited with error code.
See "systemctl status crc-dnsmasq.service" and "journalctl -xe" for details.

**Comment by praveenkumar** [comment]
@rupeshad81 Can you ssh to the VM and try to get the journalctl logs for this service? Also have you made any changes to the VM before restart or it was just `start => stop => start` ?
```
journalctl -u crc-dnsmasq.service
```

**Comment by rupeshad81** [comment]
Hi,
Please check the attached journalctl  logs.

There was no any changes made after installation, it was monitoring stage and it got failed.

Thanks
[journalctl_nodes.log](https://github.com/code-ready/crc/files/7285690/journalctl_nodes.log)

**Comment by rupeshad81** [comment]
Hi,

Please check :

[core@crc-scmfr-master-0 ~]$ journalctl -u crc-dnsmasq.service
-- Logs begin at Fri 2021-09-24 07:59:40 UTC, end at Tue 2021-10-05 12:19:56 UTC. --
Oct 05 08:01:12 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 08:01:13 crc-scmfr-master-0 podman[2268]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:01:13 crc-scmfr-master-0 systemd[1]: Started Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 08:12:51 crc-scmfr-master-0 systemd[1]: Stopping Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 08:12:52 crc-scmfr-master-0 podman[40100]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:12:52 crc-scmfr-master-0 podman[40660]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:12:53 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Succeeded.
Oct 05 08:12:53 crc-scmfr-master-0 systemd[1]: Stopped Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 08:12:53 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Consumed 1.710s CPU time
-- Reboot --
Oct 05 08:14:07 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 08:14:09 crc-scmfr-master-0 podman[1803]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:14:09 crc-scmfr-master-0 systemd[1]: Started Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 08:58:27 crc-scmfr-master-0 systemd[1]: Stopping Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 08:58:28 crc-scmfr-master-0 podman[206949]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:58:28 crc-scmfr-master-0 podman[207189]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 08:58:28 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Succeeded.
Oct 05 08:58:28 crc-scmfr-master-0 systemd[1]: Stopped Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 08:58:28 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Consumed 1.501s CPU time
-- Reboot --
Oct 05 09:00:02 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 09:00:04 crc-scmfr-master-0 podman[1753]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 09:00:04 crc-scmfr-master-0 systemd[1]: Started Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 10:13:49 crc-scmfr-master-0 systemd[1]: Stopping Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 10:13:51 crc-scmfr-master-0 podman[381324]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 10:13:51 crc-scmfr-master-0 podman[381968]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 10:13:51 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Succeeded.
Oct 05 10:13:51 crc-scmfr-master-0 systemd[1]: Stopped Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 10:13:51 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Consumed 1.421s CPU time
-- Reboot --
Oct 05 10:17:20 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 10:17:22 crc-scmfr-master-0 podman[1844]: 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
Oct 05 10:17:22 crc-scmfr-master-0 systemd[1]: Started Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
-- Reboot --
Oct 05 11:49:53 crc-scmfr-master-0 systemd[1]: Starting Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service...
Oct 05 11:50:00 crc-scmfr-master-0 podman[1764]: Error: error opening database /var/lib/containers/storage/libpod/bolt_state.db: open /var/lib/containers/storage/libpod/bolt_state.db: no >
Oct 05 11:50:00 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Control process exited, code=exited status=125
Oct 05 11:50:00 crc-scmfr-master-0 podman[2152]: Error: no container with name or ID "849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343" found: no such container
Oct 05 11:50:00 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Control process exited, code=exited status=125
Oct 05 11:50:00 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Failed with result 'exit-code'.
Oct 05 11:50:00 crc-scmfr-master-0 systemd[1]: Failed to start Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service.
Oct 05 11:50:00 crc-scmfr-master-0 systemd[1]: crc-dnsmasq.service: Consumed 371ms CPU time

**Comment by rupeshad81** [comment]
Hi Praveen,

Do we have any update to overcome this issue.


Thanks

**Comment by cfergeau** [comment]
Your issue is only 4 hours old, and it's not the only one which we have to look into, some patience is in order :)

**Comment by maratsal** [comment]
We just faced the same issue.

Basically what is happening, podman containers state file `/var/lib/containers/storage/libpod/bolt_state.db` and in addition to that crc-dnsmasq and gvisor-tap-vsock images get vanished and 2 services cannot start.

```
[systemd]
Failed Units: 2
  crc-dnsmasq.service
  gvisor-tap-vsock.service
```

Basically instead of creation of crc-dnsmasq service by using `podman generate systemd` command, those systemd services should be created with real image names instead. Here is current service details:

```
# container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service
# autogenerated by Podman 3.2.3
# Fri Sep 24 07:55:08 UTC 2021
[Unit]
Description=Podman container-849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
RequiresMountsFor=/var/run/containers/storage
[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=no
TimeoutStopSec=70
ExecStart=/usr/bin/podman start 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
ExecStop=/usr/bin/podman stop -t 10 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
ExecStopPost=/usr/bin/podman stop -t 10 849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343
PIDFile=/var/run/containers/storage/overlay-containers/849aa88cb5cd953dee534b75718babedc49dc017b74cf256c147272edb78f343/userdata/conmon.pid
Type=forking
[Install]
WantedBy=multi-user.target default.target

### gvisor service

# container-aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894.service
# autogenerated by Podman 3.2.3
# Fri Sep 24 07:55:07 UTC 2021
[Unit]
Description=Podman container-aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
RequiresMountsFor=/var/run/containers/storage
[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=no
TimeoutStopSec=70
ExecStart=/usr/bin/podman start aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894
ExecStop=/usr/bin/podman stop -t 10 aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894
ExecStopPost=/usr/bin/podman stop -t 10 aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894
PIDFile=/var/run/containers/storage/overlay-containers/aa214a4633d94a6c41c963fb00b900dcf1368ba9d08e6c7617a452904f261894/userdata/conmon.pid
Type=forking
[Install]
WantedBy=multi-user.target default.target
```

And startup should be something like with all needed options:
```
ExecStart=/usr/bin/podman run quay.io/crcont/dnsmasq:latest  
```

**Comment by berndbausch** [comment]
> this happens only if underlying host is rebooted without properly stopping crc by `crc stop` command.

Me too! My host rebooted without giving me a chance to properly shut down CRC. After this, `crc start` terminates with the crc-dnsmasq error. My version is 1.33.1+3cfaa10f.

**Comment by gbraad** [comment]
@rupeshad81 You are dealing with 'people' here that also go about their own business and work. Pushing and prodding never works, besides upsetting them. On issues we can only provide best effort 'support'. Please be patient and considered.

@maratsal Can you explain:

> those systemd services should be created with real image names instead.

This shouldn't have to matter as in this case we keep state. And your suggestion would be stateless as it would run a new instance, and leave possible instances state anyway.

The problem would be that the image or podman does not handle this gracefully to begin with. Running a new instance would be a workaround we can consider, but would like to understand the underlying problem too (as that should never happen).

**Comment by gbraad** [comment]
@vrothberg any idea what could cause the images to vanish or how to determine this?

**Comment by vimalk78** [comment]
facing the same issue. need to recreate CRC in order to use it.

**Comment by rupeshad81** [comment]
Hi,

This was monitoring stage, in which had stop the CRC and rebooted the server and started the crc, then it had failed with errors.
Also recreate the CRC is not solution for every time, 

Users had worked on it and tested the deployments for further development.

Till older crc version 4.7.X , we had never faced such issue, while doing the restart. 

In such condition , we cannot release latest crc to customer for doing the deployments.

Please can we have proper solution for it or any latest version with fixing this issue.

Thanks

**Comment by vrothberg** [comment]
> @vrothberg any idea what could cause the images to vanish or how to determine this?

It may be my morning brain but where do you see indicators that _images_ vanish?  The only thing I could see in the logs is `no such container` when the systemd unit attempts to `podman start`.

I'd encourage to have a look at `podman events --since 48h` to see which containers and images were removed.  Maybe the journal indicates who initiated it.

@mheon, have you seen something like that in the past?

**Comment by cfergeau** [comment]
Initially:
> After the crc successful installation , when we do crc stop and start , it get failed.

and just now:
> in which had stop the CRC and rebooted the server and started the crc, then it had failed with errors.

Please provide *everything* that you do in the steps to reproduce, not just some filtered data. 

If `crc stop` is successful, `crc start` works fine for me. I would guess `crc stop` is not successful/complete for you before you reboot, or that you use `crc stop -f`?

> Please can we have proper solution for it or any latest version with fixing this issue.

Now that there is a bug report with enough information to reproduce (thanks to others for providing additional information which made it possible to reproduce ;), we can look into it, and try to fix it in a future release.

**Comment by rupeshad81** [comment]
Hi,

Firstly CRC was installed successfully , it was start and stop was successfully done.

Next stage was , stop the crc with command crc  stop and then rebooted the server.

After the restart , when it was getting to start, it was showing this all errors. So I had thought of raising this issue.

As you had mentioned for next release , please can you provide the approximate date, on which it will be released.

Still now , we had not got Good CRC version of 4.8.x , So we are waiting for the earlier and also our customer waiting. There has been many escalation internally in our company for not getting good crc version 4.8.x.

So please request to get stable CRC version of 4.8.x , like we had stable version till now 4.7.x

Thanks for your co-operation.

**Comment by cfergeau** [comment]
> As you had mentioned for next release , please can you provide the approximate date, on which it will be released.

I said "in a future release", could be the next one, but it could also be a later one. We make a release every 3 weeks.

**Comment by cfergeau** [comment]
> > @vrothberg any idea what could cause the images to vanish or how to determine this?
> 
> It may be my morning brain but where do you see indicators that _images_ vanish? The only thing I could see in the logs is `no such container` when the systemd unit attempts to `podman start`.
> 
> I'd encourage to have a look at `podman events --since 48h` to see which containers and images were removed. Maybe the journal indicates who initiated it.
> 
> @mheon, have you seen something like that in the past?

Looking some more into it, I can see this in the logs:
```
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.727872537Z" level=info msg="Starting CRI-O, version: 1.21.3-2.rhaos4.8.gitff0b7fe.el8, git: ()"
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.862511748Z" level=info msg="Checking whether cri-o should wipe containers: version file /var/run/crio/version not found: open /var/run/crio/version: no such file or directory"
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.862672576Z" level=info msg="File /var/lib/crio/clean.shutdown not found. Wiping storage directory /var/lib/containers/storage because of suspected dirty shutdown"
```

which is from 

```
$ cat /usr/lib/systemd/system/crio-wipe.service
[Unit]
Description=CRI-O Auto Update Script
Before=crio.service
RequiresMountsFor=/var/lib/containers

[Service]
EnvironmentFile=-/etc/sysconfig/crio
ExecStart=/usr/bin/crio \
          $CRIO_CONFIG_OPTIONS \
          $CRIO_RUNTIME_OPTIONS \
          $CRIO_STORAGE_OPTIONS \
          $CRIO_NETWORK_OPTIONS \
          $CRIO_METRICS_OPTIONS \
          wipe

Type=oneshot

[Install]
WantedBy=multi-user.target
```

Probably nothing directly related to podman, just the image we are using which is not behaving as we want it to :)

**Comment by gbraad** [comment]
> We make a release every 3 weeks.

We try to keep a cadence of 3 weeks, but not a given. especially seen we have so many people being on leave. The following releases might only address critical fixes.

**Comment by gbraad** [comment]
@cfergeau thanks for the thorough investigation! But seems like a nasty side-effect to wipe all container images.

**Comment by rupeshad81** [comment]
Hi,

As per your comments, it release every 3 weeks , agreed.

But in this version ,it has not able get the stable crc, then in this case, fix or workaround should be given.
or Need to release the new version as early as possible, So that we can use it.

Please provide the solution as early as possible.

Thanks

**Comment by vrothberg** [comment]
Thanks for checking, @cfergeau!

```
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.727872537Z" level=info msg="Starting CRI-O, version: 1.21.3-2.rhaos4.8.gitff0b7fe.el8, git: ()"
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.862511748Z" level=info msg="Checking whether cri-o should wipe containers: version file /var/run/crio/version not found: open /var/run/crio/version: no such file or directory"
Oct 06 11:37:27 crc-scmfr-master-0 crio[1012]: time="2021-10-06 11:37:27.862672576Z" level=info msg="File /var/lib/crio/clean.shutdown not found. Wiping storage directory /var/lib/containers/storage because of suspected dirty shutdown"
```

> @cfergeau thanks for the thorough investigation! But seems like a nasty side-effect to wipe all container images.

I concur.  CRI-O has detected an ungraceful shutdown and is nuking the storage.  That is OK in a Kubernetes world but is conflicting with Podman (which handles storage corruptions at runtime) when both share the same storage.

I see two possible solutions at the moment:
1) Have CRI-O and Podman have different storage paths
2) Change the systemd units such that the containers are not merely started but created. See `podman generate systemd --new`.  With `--new`, the simple `podman start` is changed to `podman run` along with a number of other things.  This way, the image would get pulled on demand. It depends on the kind of workload the container runs but I would suspect that the container could be recreated at boot. Some more background in older blog: https://www.redhat.com/sysadmin/podman-shareable-systemd-services

@mrunalp @rhatdan FYI

**Comment by cfergeau** [comment]
> I concur. CRI-O has detected an ungraceful shutdown and is nuking the storage. That is OK in a Kubernetes world but is conflicting with Podman running in these specific systemd units when both share the same storage.

Yeah, cri-o people also agree it's suboptimal :) https://github.com/cri-o/cri-o/commit/086982d6184f73681999828905b5a422a92f7024

**Comment by gbraad** [comment]
So now wondering when this would be part of the image. The alternative would be to disable/mask the unit on startup?

**Comment by vrothberg** [comment]
That sounds like solution 3 which seems easy and straight forward.  Option 2 may be the most robust long term.

**Comment by gbraad** [comment]
Maybe this is the root cause behind: https://github.com/code-ready/crc/issues/2678

**Comment by rupeshad81** [comment]
Hi,

Do we any update on the crc stable?

Please have update on it.

Thanks

**Comment by gbraad** [comment]
A solution has been added to the SNC generation scripts and therefore the fix will be part of the next release.

**Comment by gbraad** [comment]
@rupeshad81 A new release has been made available: https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/crc/1.34.0 Do not, that although the fix is in, this also uses a new version of OpenShift 4.9. If you experience issues, please consider opening a new issue.
